platforms/common/include/px4_platform_common/crypto_backend.h

```diff
@@ -327,6 +369,16 @@ typedef struct cryptoiocdecryptdata {
 	size_t ret;
 } cryptoiocdecryptdata_t;
 
+#define CRYPTOIOCGENKEYPAIR _CRYPTOIOC(12)
+typedef struct cryptoiocgenkeypair {
+	crypto_session_handle_t *handle;
+	size_t key_size;
+	uint8_t private_idx;
```

Comment:
What was the reason again not to store private and public parts together with the same index?

Reply:
It is using only one key index (the private one) underneath, but I didn't bother to touch the upstreamed API. But, I can change that also and remove the other idx

Reply:
well.. the sw_crypto impl I made does not use one slot.. need to fix that.